sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/join/CoGbkResult.java

```diff
@@ -59,6 +60,8 @@
 
   private static final int DEFAULT_IN_MEMORY_ELEMENT_COUNT = 10_000;
 
+  private static final int DEFAULT_MIN_ELEMENTS_PER_TAG = 100;
```

Comment:
Based upon the code it looks like this is used as a max per tag.

Reply:
The idea is that we will cache always at least this many values per tag, regardless of whether DEFAULT_IN_MEMORY_ELEMENT_COUNT was "used up" for other tags. I'll clarify.